xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -202,56 +198,20 @@ public void reset() {
         localityMap.get(locality).shutdown();
       }
       localityMap.clear();
-
-      for (Locality locality : localities) {
-        loadStatsStore.removeLocality(locality);
-      }
-      localities = ImmutableSet.of();
-
       priorityManager.reset();
     }
 
     @Override
     public void updateLocalityStore(final Map<Locality, LocalityLbEndpoints> localityInfoMap) {
-
       Set<Locality> newLocalities = localityInfoMap.keySet();
       // TODO: put endPointWeights into attributes for WRR.
       for (Locality locality : newLocalities) {
         if (localityMap.containsKey(locality)) {
-          LocalityLbInfo localityLbInfo = localityMap.get(locality);
-          LocalityLbEndpoints localityLbEndpoints = localityInfoMap.get(locality);
-          handleEagsOnChildBalancer(helper, localityLbInfo, localityLbEndpoints, locality);
```

Comment:
BTW, this was also a bug. You should pass in the  `ChildHelper` instance of the `LocalityLbInfo` instead of just `helper`. It's lucky that `ChildHelper`'s `SynchronizationContext` is just a delegation to `helper`'s, so it did not cause problem. But this is wrong.

Argument-passing, object ownership, lifecycle, etc. A lot of things in `LocalityStore` are messy. I hate this class, really. It might still contain unnoticed bug. Hope we can refactor to the LB structure and get rid of this early.

Reply:
I think we should not hold a counter ref in `ChildHelper`, instead call `loadStatsStore.getLocalityCounter(locality)` whenever `ChildHelper` need use the counter. `LocalityStore` does not know when `LoadStatsStore` do the magic.

Reply:
It should not pass any helper now. It should pass a `SynchronizationContext`. Helper's SynchronizationContext should guarantee it works properly with load balancer's synchronization.

Reply:
A lot of argument-passing, object ownership, lifecycle issues are messed up by load stats. Spec of both EDS and load reporting are constantly changing, it's just unfortunate.

Reply:
> It should not pass any helper now. It should pass a `SynchronizationContext`. Helper's SynchronizationContext should guarantee it works properly with load balancer's synchronization.

I refactored it to not do that.

Things were just working fortunately correct. But the idea and the way it was coded is wrong.